cornichon-core/src/main/scala/com/github/agourlay/cornichon/http/HttpRequest.scala

```diff
@@ -128,5 +129,16 @@ case class QueryGQL(url: String, query: Document, operationName: Option[String]
     val vars: Map[String, Json] = newVariables.map { case (k, v) ⇒ k → parseJsonUnsafe(v) }(scala.collection.breakOut)
     copy(variables = variables.fold(Some(vars))(v ⇒ Some(v ++ vars)))
   }
+
+  lazy val payload: String = {
+    import io.circe.generic.auto._
+    import io.circe.syntax._
+
+    val querySource = query.source.getOrElse(QueryRenderer.render(query, QueryRenderer.Compact))
```

Comment:
Well in `QueryGQL` the `payload` is the whole json object.
in `queryGqlToStep` `prettyPayload` is just the query part.

Since prettifying is not a concern of `QueryGQL` I didn't want to expose another field.
But I see that parsing it twice is a high price.